crates/criticalup-core/src/config/paths.rs

```diff
@@ -40,6 +40,19 @@ impl Paths {
                 find_cache_dir(whitelabel).ok_or_else(|| Error::CouldNotDetectCacheDirectory)?
             }
         };
+        let old_path = cache_dir.join("artifacts").join("ferrocene");
+        let new_path_parent = cache_dir.join("artifacts").join("products");
+        // We want to do the migration only if the new path was never created
+        // because we do not want to overwrite it if it already exists.
+
+        // We want to access and check that we have permission.
+
+        if old_path.try_exists().is_ok_and(|_| true)
+            && new_path_parent.try_exists().is_ok_and(|_| false)
+        {
+            migrate_obsolete_path(old_path.clone(), new_path_parent.clone())
```

Comment:
```suggestion
            migrate_deprecated_path(old_path.clone(), new_path_parent.clone())
```

unify naming "old"  and 'obsolete' to 'deprecated'